### PR TITLE
feat(inputdropdown): implement new redesigned inputdropdown

### PR DESCRIPTION
**Pull request by masoudmanson**
## Summary

**InputDropdown**
Github issue: #437 

## Checklist

- [x] Default Story in Storybook
- [x] LivePreview Story in Storybook
- [x] Test Story in Storybook
- [x] Tests written
- [x] Variables from `defaultTheme.ts` used wherever possible
- [x] If updating an existing component, depreciate flag has been used where necessary
- [ ] Chromatic build verified by @chanzuckerberg/sds-design

### CODE REVIEW

**Review by clarsen-czi — DISMISSED** [review]
Looking really good, @masoudmanson !! Just a few changes to make.

File: packages/components/src/core/InputDropdown/style.ts [hunk 1]
```diff
@@ -20,9 +21,11 @@ export interface InputDropdownProps extends CommonThemeProps {
   open?: boolean;
   sdsStage: "default" | "userInput";
   sdsStyle?: "minimal" | "square" | "rounded";
-  sdsType?: "singleSelect" | "multiSelect";
+  sdsType?: "label" | "value";
```
Comment (tihuan) [inline packages/components/src/core/InputDropdown/style.ts:24]: looks like somewhere in this file we need to filter out `sdsType` from passing to the DOM 😄 
Screenshot 2023-07-19 at 12 55 58 PM

Comment (masoudmanson) [reply packages/components/src/core/InputDropdown/style.ts:24]: Ahh, I just fixed it :D

File: packages/components/src/core/InputDropdown/style.ts [hunk 2]
```diff
@@ -20,9 +21,11 @@ export interface InputDropdownProps extends CommonThemeProps {
...
+  multiple?: boolean;
   details?: string;
   counter?: string;
```
[the remark this answers is not in the document]
Comment (masoudmanson) [reply packages/components/src/core/InputDropdown/style.ts:27]: Sure thing, that sounds great!

File: packages/components/src/core/InputDropdown/index.tsx [hunk 3]
```diff
@@ -51,53 +53,101 @@ const InputDropdown = (props: InputDropdownProps): JSX.Element => {
...
+  details: InputDropdownProps["details"];
+  isMinimal: boolean;
+  label: InputDropdownProps["label"];
+  multiple: InputDropdownProps["multiple"];
+  shouldPutAColonAfterLabel: InputDropdownProps["shouldPutAColonAfterLabel"];
+  sdsType: InputDropdownProps["sdsType"];
+  value: InputDropdownProps["value"];
+}
+
+// eslint-disable-next-line sonarjs/cognitive-complexity
```
Comment (tihuan) [inline packages/components/src/core/InputDropdown/index.tsx:107]: The following suggestion is a way to reduce complexity!

```ts
function renderLabelText(props: RenderLabelTextProps) {
  const { sdsType } = props;

  if (sdsType === "label") {
    return renderLabelTypeLabelText(props);
  }

  if (sdsType === "value") {
    return renderValueTypeLabelText(props);
  }

  return null;
}

function renderLabelTypeLabelText({
  isMinimal,
  label,
  counter,
  value,
  shouldPutAColonAfterLabel,
}: RenderLabelTextProps) {
  if (isMinimal) return label;

  return (counter || value) && shouldPutAColonAfterLabel ? `${label}:` : label;
}

function renderValueTypeLabelText({
  value,
  multiple,
  isMinimal,
  label,
  counter,
  details,
  shouldPutAColonAfterLabel,
}: RenderLabelTextProps) {
  if (!value || multiple) {
    if (isMinimal) return label;

    return counter && shouldPutAColonAfterLabel ? `${label}:` : label;
  }

  if (isMinimal) return value;

  return (counter || details) && shouldPutAColonAfterLabel
    ? `${value}:`
    : value;
}
```

**Review by tihuan — APPROVED** [review]
LGTM!! Thanks for the awesome work on this, @masoudmanson 🙏 🙏 🙏 

Just some minor feedback!

File: packages/components/src/core/InputDropdown/style.ts (hunk 1 above)
Comment (tihuan) [reply packages/components/src/core/InputDropdown/style.ts:24]: hurray you the best!!

File: packages/components/src/core/InputDropdown/index.tsx (hunk 3 above)
Comment (masoudmanson) [reply packages/components/src/core/InputDropdown/index.tsx:107]: Oh woow, Thank you very much @tihuan 
It is much better this way 😯🤩

**Pull request merged**